docs/wip/data-fetching.md

```diff
@@ -1,8 +1,58 @@
+
 ### Static data
+
+`Static data` is the content that does not change frequently or per user interaction.
+Considering that, you can use the fetching during building time for data that don't change very often, making the storefront loading faster.
+
+For example, a product's page can be a data retrieve static since the description and attributes of a product rarely chages. This means that this data can be fetch during build time and when the user access the products page the data is already loaded.
+
+There are two ways for fetch static data in Gatsby:
+  - [Page Query](https://www.gatsbyjs.com/docs/recipes/querying-data#querying-data-with-a-page-query)
+  - [Static Query](https://www.gatsbyjs.com/docs/how-to/querying-data/static-query/#reach-skip-nav).
+
 ### Dynamic data
 
-### What we provided out of the box with `gatsby-source-vtex`
+If the data constantly changes or depends on the user's interaction, make it to be dynamic fetched.
+
+For example: when a shipping calculation changes depending on the user's location or when implementing a shopping cart that display diferrent content every time a product is added, demanding backend communcation at every user's change — use the dynamic.
+
+For fetching data at runtime in the browser, you can apply any method to retrieve data that you would use in a regular React app, such as the `fetch` API for retrivieng data of a Rest API or `apollo-client` to retrieve from a GraphQL API.
+
+## What we provide with `gatsby-source-vtex` that is out of the box
+
+`gatsby-source-vtex` plugin connects the [Gatsby data layer](https://www.gatsbyjs.com/docs/porting-from-create-react-app-to-gatsby/#unified-graphql-data-layer) to [VTEX's GraphQL](https://github.com/vtex-apps/store-graphql) allowing it to access VTEX API through your Gatsby application.
+
+### Plugin Setup
+
+For the plugin setup, go the `gatsby-config.js` file and change the `STORE_ID` variable to the desired VTEX account.
+
+It is also possible to edit the variables `enviroment` and `workspace`, example:
+
+```js
+plugins: [
+    {
+      resolve: `@vtex/gatsby-source-vtex`,
+      options: {
+        tenant: STORE_ID, // VTEX's account
+        environment: 'vtexcommercestable' , // it can be vtexcommercestable or vtexcommercebeta
+        workspace: 'master', // https://vtex.io/docs/concepts/workspace/ 
+      },
+    },
+```
+
+Run the site with the `yarn develop` command, than check if the output at the end of the running summary contains the URL to access Gatsby's GraphQl playground:
+
+```
+View GraphiQL, an in-browser IDE, to explore your site data and schema
+⠀
+http://localhost:8000/___graphql
+```
+
+Access this URL and to see all queries and mutations of the VTEX API in the Schema.
+
+Feel free to test the queries and see what they return. To use this data on your website, just follow what is indicated in the previous topic of [How to fetch data for your pages](#How-to-fetch-data-for-your-pages).
```

Comment:
```suggestion
To be able to use this plugin, it's necessary to configure its settings. Take the following steps:

1. Open your SFJ project in any code editor of your choosing.
2. From the root directory, go to the `gatsby-config.js` file.
3. Head to `plugins` and, inside `@vtex/gatsby-source-vtex`, replace the `STORE_ID` value with the `name` of your VTEX account.

>ℹ️ *The [`gatsby-config.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/) file defines your site’s metadata, plugins, and other general configurations.*

    ```js
    plugins: [
        {
          resolve: `@vtex/gatsby-source-vtex`,
          options: {
            tenant: STORE_ID, // VTEX account name
            environment: 'vtexcommercestable' , // it can be vtexcommercestable or vtexcommercebeta
            workspace: 'master', // https://vtex.io/docs/concepts/workspace/ 
          },
        },
    ```

4. Run the site with the `yarn develop` command. At the end of the running summary, you'll receive a successful message with the URL to the [Gatsby GraphQl playground](https://www.gatsbyjs.com/docs/using-graphql-playground/#reach-skip-nav).

    ```
    View GraphiQL, an in-browser IDE, to explore your site data and schema
    ⠀
    http://localhost:8000/___graphql
    ```

Now, if you access this URL in your browser, you'll be able to fetch and change data from the [Store GraphQL interface](https://github.com/vtex-apps/store-graphql). Feel free to test the queries and see what they return. 

To use this data on your website, follow the referenced links in the [How to fetch data for your pages](#How-to-fetch-data-for-your-pages) section.
```